Add toggle button

jrmoulton opened the pull request:
This works but could definitely be cleaned up. @Zoxc this is the first time I'm using the style classes/props and I'd appreciate any feedback. Especially if I'm missing the point. 

There are sizing-related things that are set inside of `paint` which feels weird to me but I couldn't figure out where else to access the properties. 

A styled example. 
![image](https://github.com/lapce/floem/assets/13090441/142513b8-ff68-4eb8-9ec6-9e085011afef)

dzhou121 commented:
Looks beautiful!

One thing though is maybe we want to use an event listener style for the toggle value (```on_change``` or ```on_toggle```) rather than a signal. (And change checkbox and text input to use event listener style as well, but that can be in a different PR)

Zoxc commented on src/views/toggle_button.rs:
```diff
@@ -0,0 +1,151 @@
+
+prop!(pub ToggleBg: Option<peniko::Color> {} = None);
+prop!(pub ToggleFg: Option<peniko::Color> {} = None);
+style_class!(pub ToggleClass);
```
`ToggleButtonClass` to match?

Zoxc commented on src/views/toggle_button.rs:
```diff
@@ -0,0 +1,151 @@
+use super::Decorators;
+
+prop!(pub ToggleBg: Option<peniko::Color> {} = None);
+prop!(pub ToggleFg: Option<peniko::Color> {} = None);
```
I'd add a builtin `Foreground foreground nocb: Option<Color> {} = None,` property and use that along with `Background`. They should be read in the `style` pass, see `label.rs`.

jrmoulton replied:
The issue with using `Background` is that painting the background is a special case that is handled in `paint_view`. And that would require that the user always set a border radius. Which could be done in a default style?

jrmoulton replied:
But that border radius needs be relative to the size of the element which isn't super convenient to set as a user

Zoxc commented on src/views/toggle_button.rs:
```diff
@@ -0,0 +1,151 @@
+        let layout = cx.get_layout(self.id).unwrap();
+        let size = Size::new(layout.size.width as f64, layout.size.height as f64);
+
+        self.width = size.width;
```
This seems like it belongs in `compute_layout`.

jrmoulton commented:
Added custom inset and size properties

![image](https://github.com/lapce/floem/assets/13090441/8eceaf46-5b72-4def-886e-f8583252f461)

jrmoulton commented:
Should be good to go

jrmoulton commented:
I've switched the button to use the regular background and added a built-in foreground style. I've also made it so that the border radius takes a PxPct so that a percentage border radius can be set.
